packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -421,12 +421,8 @@ export class ProjectModel {
         return cachedExplores;
     }
 
-    async tryAcquireProjectLock(
-        projectUuid: string,
-        onLockAcquired: () => Promise<void>,
-        onLockFailed?: () => Promise<void>,
-    ): Promise<void> {
-        await this.database.transaction(async (trx) => {
+    async tryAcquireProjectLock(projectUuid: string): Promise<boolean> {
```

Comment:
Removed callbacks since there is no advantage in do in it. This also allow us to have cleaner code in the service.